src/ploigos_step_runner/step_implementers/create_container_image/maven_jkube_k8sbuild.py

```diff
@@ -0,0 +1,212 @@
+    def step_implementer_config_defaults():
+        """Getter for the StepImplementer's configuration defaults.
+
+        Returns
+        -------
+        dict
+            Default values to use for step configuration values.
+
+        Notes
+        -----
+        These are the lowest precedence configuration values.
+        """
+        return {**MavenGeneric.step_implementer_config_defaults(), **DEFAULT_CONFIG}
+
+    @staticmethod
+    def _required_config_or_result_keys():
+        """Getter for step configuration or previous step result artifacts that are required before
+        running this step.
+
+        See Also
+        --------
+        _validate_required_config_or_previous_step_result_artifact_keys
+
+        Returns
+        -------
+        array_list
+            Array of configuration keys or previous step result artifacts
+            that are required before running the step.
+        """
+        return REQUIRED_CONFIG_OR_PREVIOUS_STEP_RESULT_ARTIFACT_KEYS
+
+    def _run_step(self): # pylint: disable=too-many-locals
+        """Runs the step implemented by this StepImplementer.
+
+        Returns
+        -------
+        StepResult
+            Object containing the dictionary results of this step.
+        """
+        step_result = StepResult.from_step_implementer(self)
+
+        # create local build tag
+        image_registry_organization = self.get_value('organization')
+        build_full_tag, build_short_tag, image_registry_uri, image_repository, image_version = \
+            determine_container_image_build_tag_info(
+                image_version=self.get_value('container-image-version'),
+                organization=image_registry_organization,
+                application_name=self.get_value('application-name'),
+                service_name=self.get_value('service-name')
+            )
+
+        # push the artifacts
+        mvn_jkube_output_file_path = self.write_working_file('mvn_k8s_build_output.txt')
+        try:
+            # execute maven step (params come from config)
+            print("Build container image with Maven Jkube kubernetes plugin")
+            self._run_maven_step(
+                mvn_output_file_path=mvn_jkube_output_file_path,
+                step_implementer_additional_arguments=[
+                    f"-Djkube.generator.name={build_full_tag}"
+                ]
+            )
+        except StepRunnerException as error:
+            step_result.success = False
+            step_result.message = "Error running 'maven k8s:build' to create container image. " \
+                f"More details maybe found in 'maven-jkube-output' report artifact: {error}"
+        finally:
+            step_result.add_artifact(
+                description="Standard out and standard error from running maven to " \
+                    "create container image.",
+                name='maven-jkube-output',
+                value=mvn_jkube_output_file_path
+            )
+
+        # add artifacts
```

Comment:
Wish there was a way to DRY this up but KISS is always better, right?

Reply:
indeed.